code/obj/machinery/manufacturer.dm

```diff
@@ -542,7 +542,7 @@ TYPEINFO(/obj/machinery/manufacturer)
 							src.mend(usr, text2num_safe(params["wire"]))
 						return TRUE
 				if ("pulse")
-					if (!ispulsingtool(usr.equipped()) || (isAI(usr) || isrobot(usr)))
+					if (!ispulsingtool(usr.equipped()) && !(isAI(usr) || isrobot(usr)))
```

Comment:
What's the reason AI/robot matters here? If they don't have a pulsing tool, they don't have a pulsing tool, right?

Reply:
AI/Robot have always been able to pulse long-range IIRC. AIs wouldnt be able to do it otherwise since they dont get one